### PR TITLE
docs: Update Development and CONTRIBUTING docs

### PR DESCRIPTION
**Pull request by mamoodi**
This doesn't exactly resolve this https://github.com/OpenDevin/OpenDevin/issues/2386 as described, however, it does resolve the main issue which is informing the user of each document. I first tried to merge them but it didn't look good TBH.

Notable Changes:
* Updated both Development.md and CONTRIBUTING.md beginning sentences to make user aware that each doc exists
* Formatting and consistency changes for both documents
* Some additional changes in Development.md regarding configuration variable precedence

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -40,24 +39,24 @@ git push origin main
 ```
 
 ### 4. Create a New Branch And Open a Pull Request
-After you finish implementation, open forked repository. The source branch is your new branch, and the target branch is `OpenDevin/OpenDevin` `main` branch. Then PR should appears in [OpenDevin PRs](https://github.com/OpenDevin/OpenDevin/pulls).
+Create a new branch on your fork with your implementation. Open your forked repository and create a Pull Request which sets the source branch to your new branch, and the target branch to `OpenDevin/OpenDevin` `main` branch.
```
Comment (tobitege) [inline CONTRIBUTING.md:42]: For non-English speakers this sentence is hard to parse at first glance, not everyone is a major in Git-ology. :)
For many it is obvious, but not every developer uses Git daily.
Maybe a small screenshot can be a helpful illustration.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -67,24 +66,21 @@ For example, a PR title could be:
 
 You may also check out previous PRs in the [PR list](https://github.com/OpenDevin/OpenDevin/pulls).
 
-As described in [here](https://github.com/OpenDevin/OpenDevin/labels), we create several labels. Every PR should be tagged with the corresponding labels.
+As described [here](https://github.com/OpenDevin/OpenDevin/labels), we have created several labels. Every PR should be tagged with the corresponding labels.
```
Comment (tobitege) [inline CONTRIBUTING.md:69]: I have no idea myself yet how to add a label, speaks to the UI of Github 🤣

File: Development.md [hunk 3]
```diff
@@ -27,42 +28,41 @@ mamba install conda-forge::poetry
...
-   With a single command, you can have a smooth LM setup for your OpenDevin experience. Simply run:
+To configure the LM of your choice, run:
+       
    ```bash
    make setup-config
    ```
+   
    This command will prompt you to enter the LLM API key, model name, and other variables ensuring that OpenDevin is tailored to your specific needs. Note that the model name will apply only when you run headless. If you use the UI, please set the model in the UI.
-   Set `persist_sandbox` to false if you want to use clean sandbox for each task. If `persist_sandbox` is set to true, you will need to set the `ssh_password` as well.
+   Set `persist_sandbox` to false if you want to use a clean sandbox for each task. If `persist_sandbox` is set to true, you will need to set the `ssh_password` as well.
```
Comment (tobitege) [inline Development.md:47]: Might be good for now to mention, that persist_sandbox=true is a bit problematic.

File: CONTRIBUTING.md (hunk 1 above)
Comment (mamoodi) [reply CONTRIBUTING.md:42]: That's great feedback. Let me see how I can make it more clear and not require a Git major haha.

Comment (mamoodi) [reply CONTRIBUTING.md]: How does it look now?

Comment (tobitege) [reply CONTRIBUTING.md]: Reads great  now!

**Review by tobitege — APPROVED** [review]
Great job! Thanks for looking into the details!
